app/store/StoreAccessor.ts

```diff
@@ -15,6 +43,29 @@ class StoreAccessor {
   getStore = () => {
     return this.store;
   };
+
+  initialize = (withIntersectMiddleware: boolean): [Store, Persistor] => {
```

Comment:
Took a hack at moving the initialization here, right now is just a flag but I think we could extend it to a config object, particular to each strategy if the requirements change. Right now it just diverges on the one middleware so having a flag here looked okayish. I think it is an improvement from my first shot at it. Sorry to put you all through this but your input is amazing. So I am selfishly soliciting more of it. 😄